pkgdb/include/flox/pkgdb/read.hh

```diff
@@ -40,6 +41,21 @@
 /** @brief Interfaces for caching package metadata in SQLite3 databases. */
 namespace flox::pkgdb {
 
+/* -------------------------------------------------------------------------- */
+
+const DurationMillis DB_RETRY_PERIOD = DurationMillis( 100 );
+
+#define RETRY_WHILE_BUSY( op )                        \
+  int _retry_while_busy_rcode   = op;                 \
+  int _retry_while_busy_retries = 0;                  \
+  while ( ( _retry_while_busy_rcode == SQLITE_BUSY )  \
+          && ( ++_retry_while_busy_retries < 100 ) )  \
```

Comment:
suggestion blocking: looks like the max wait time here is 10 seconds, which I think could be shorter than a scrape. If the goal is to let the first scrape finish, I think this should be longer (and the reasoning should be explained in a comment)

Reply:
Good point, I've set the timeout to be 100 seconds and added a `throw` if we hit the maximum number of retries rather than silently continuing.